secure-storage/secure-storage-impl/src/main/java/com/duckduckgo/securestorage/impl/encryption/EncryptionHelper.kt

```diff
@@ -69,66 +54,39 @@ interface EncryptionHelper {
 
 @ContributesBinding(AppScope::class)
 class RealEncryptionHelper @Inject constructor() : EncryptionHelper {
-    private val cipher = Cipher.getInstance(TRANSFORMATION)
+    private val encryptionCipher = Cipher.getInstance(TRANSFORMATION)
+    private val decryptionCipher = Cipher.getInstance(TRANSFORMATION)
```

Comment:
The fix I implemented has 3 parts:

1. I created a separate Cipher for encryption and decryption. This way we avoid the risk of encryption and decryption flow affecting each other since it is possible that we encrypt and decrypt at almost the same time.
2. I reduced the functions to just 1 encrypt and 1 decrypt functions. I delegated the string to byteArray transformation (vice versa) out of the EncryptionHelper.
3. To complete the fix, i synchronised the encrypt and decrypt functions so we know that only one encryption or one decryption process takes place at a time.

Reply:
Sounds good to me, and thanks for the detailed explanation of the changes. 

My initial thought on an alternative approach would be to move the cipher creation inside the functions, so that they were local (and we wouldn't need to worry about synchronization), but that would be bad if `Cipher.getInstance(TRANSFORMATION)` was expensive to perform. I suspect it might be; do you happen to know?

I'm happy with the approach in the PR, but want to throw that in for consideration.

Reply:
Will address this in a separate PR. For now, we keep it this way.